### ISSUE TITLE
Add import source in metafile.inputs

### ISSUE THREAD

**Issue by sanyuan0704** [evanw/esbuild#2262]
Through the current metafile, we can analyze the module dependency graph, but we do not know how a specific dependency is imported. For example, `import React from 'react'`, then the original import path is react. Of course, there are many submodules imported. Hopefully this information can be included in esbuild's metafile, for example:
```ts
interface Metadata {
  inputs: {
    [path: string]: {
      bytes: number
      imports: {
        path: string
        kind: string
        // raw request path
        request: string
      }[]
    }
  }
}
```

**Comment by hardfist** [comment]
seems same as https://github.com/evanw/esbuild/issues/1694

**Comment by evanw** [comment]
By `raw request path` it sounds like you're asking for the original (pre-resolve) import path instead of the calculated (post-resolve) import path. Is that correct?

For example, in the following code:

```js
import React1 from 'react'
import React2 from './node_modules/react/index.js'
```

Right now the metafile contains this:

```json
"index.js": {
  "bytes": 78,
  "imports": [
    {
      "path": "node_modules/react/index.js",
      "kind": "import-statement"
    },
    {
      "path": "node_modules/react/index.js",
      "kind": "import-statement"
    }
  ]
}
```

Are you asking for something like this?

```json
"index.js": {
  "bytes": 78,
  "imports": [
    {
      "path": "node_modules/react/index.js",
      "kind": "import-statement",
      "request": "react"
    },
    {
      "path": "node_modules/react/index.js",
      "kind": "import-statement",
      "request": "./node_modules/react/index.js"
    }
  ]
}
```

**Comment by sanyuan0704** [comment]
Yes，that's right.



---Original---
From: "Evan ***@***.***&gt;
Date: Fri, May 27, 2022 03:30 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [evanw/esbuild] Add import source in metafile.inputs (Issue#2262)




 
By raw request path it sounds like you're asking for the original (pre-resolve) import path instead of the calculated (post-resolve) import path. Is that correct?
 
For example, in the following code:
 import React1 from 'react' import React2 from './node_modules/react/index.js'
 
Right now the metafile contains this:
 "index.js": {   "bytes": 78,   "imports": [     {       "path": "node_modules/react/index.js",       "kind": "import-statement"     },     {       "path": "node_modules/react/index.js",       "kind": "import-statement"     }   ] }
 
Are you asking for something like this?
 "index.js": {   "bytes": 78,   "imports": [     {       "path": "node_modules/react/index.js",       "kind": "import-statement",       "request": "react"     },     {       "path": "node_modules/react/index.js",       "kind": "import-statement",       "request": "./node_modules/react/index.js"     }   ] }

**Comment by evanw** [comment]
This has recently been implemented in [version 0.16.13](https://github.com/evanw/esbuild/releases/tag/v0.16.13). I called it `original` instead of `request` but it's otherwise the same.
